File: openapi-test-java-server-spring-client-jdk/testapi/model/Currency.java
package testapi.model;

import com.fasterxml.jackson.annotation.JsonValue;

/** ISO 4217 currency code */
public record Currency(@JsonValue String value) {
  public Currency withValue(String value) {
    return new Currency(value);
  };

  @Override
  public String toString() {
    return value;
  };

  /** Converts a String to this type for Spring @PathVariable binding */
  static public Currency valueOf(String s) {
    return new Currency(s);
  };
}

File: openapi-test-kotlin-server-spring-client-jdk/testapi/model/Currency.kt
package testapi.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

/** ISO 4217 currency code */
data class Currency @JsonCreator constructor(@get:JsonValue val value: String) {
  override fun toString(): String {
    return value
  }

  companion object {
    /** Converts a String to this type for Spring @PathVariable binding */
    fun valueOf(s: String): Currency = Currency(s)
  }
}